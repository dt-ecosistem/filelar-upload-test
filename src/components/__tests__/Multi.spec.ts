import { describe } from 'vitest'
import { mount } from '@vue/test-utils'
import FileUploader from '../MultipleUpload.vue'

describe('FileUploader', test => {
  test('should render a file input', () => {
    const wrapper = mount(FileUploader)
    const inputElement = wrapper.find('input[type="file"]').element as HTMLInputElement
    const file = new File(["foo"], "foo.txt", {
      type: "text/plain",
    });
    const mockFileList = Object.create(inputElement.files);
    mockFileList[0] = file;
    Object.defineProperty(mockFileList, "length", { value: 1 });

    (wrapper.getCurrentComponent().exposed as unknown as any).onFileChange({target: {files: mockFileList}});

  })

})
