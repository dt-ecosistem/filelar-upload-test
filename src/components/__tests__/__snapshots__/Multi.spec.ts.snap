// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MultipleUpload > check to snapshot SingleUpload 1`] = `
"<hr>
<div id="main" class="flex flex-wrap w-[100%] gap-2 p-2 mt-3.5 p-4"><input type="file" class="hidden" multiple="" id="multiple"><!-- Selected files -->
  <!-- First input --><label for="multiple" class="w-[300px] h-[250px] flex flex-col justify-center items-center cursor-pointer bg-gray-200 rounded text-white border-dashed border-2 border-gray-400"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="80" height="80" id="upload-in-cloud">
      <defs>
        <path id="a" d="M.06 64V.04H64V64z"></path>
      </defs>
      <g fill="none" fill-rule="evenodd">
        <mask id="b" fill="#fff">
          <use xlink:href="#a"></use>
        </mask>
        <path fill="#FFFFFE" d="M48 30c0-8.837-7.164-16-16-16-7.498 0-13.773 5.165-15.508 12.126C11.687 26.854 8 30.991 8 36c0 5.523 4.477 10 10 10h8a2 2 0 0 0 2-2v-5.5a1.5 1.5 0 0 0-1.5-1.5h-3.061c-1.472 0-1.883-1.01-.914-2.245l7.713-9.829c.969-1.235 2.555-1.235 3.524 0l7.713 9.829c.969 1.235.558 2.245-.914 2.245H37.5a1.5 1.5 0 0 0-1.5 1.5V44a2 2 0 0 0 2 2h10a8 8 0 0 0 0-16" mask="url(#b)"></path>
        <path fill="#4C5665" d="M48 46H38a2 2 0 0 1-2-2v-5.5a1.5 1.5 0 0 1 1.5-1.5h3.061c1.472 0 1.883-1.01.914-2.245l-7.713-9.829c-.969-1.235-2.555-1.235-3.524 0l-7.713 9.829c-.969 1.235-.558 2.245.914 2.245H26.5a1.5 1.5 0 0 1 1.5 1.5V44a2 2 0 0 1-2 2h-8c-5.523 0-10-4.477-10-10 0-5.009 3.687-9.146 8.492-9.874C18.227 19.165 24.502 14 32 14c8.836 0 16 7.163 16 16a8 8 0 0 1 0 16zM32 0C14.327 0 0 14.327 0 32c0 17.673 14.327 32 32 32 17.673 0 32-14.327 32-32C64 14.327 49.673 0 32 0z" mask="url(#b)"></path>
      </g>
    </svg>
    <p class="font-bold text-gray-500">Click to upload</p><img class="max-w-[100%] max-h-[100%]">
  </label><button type="submit" class="h-[50px] w-[100px] ml-5 p-3 bg-gray-600 rounded text-white rounded-2xl"> Upload </button>
</div>"
`;
